tests/lib/bin/session-tool

```diff
@@ -0,0 +1,110 @@
+#!/bin/sh -e
+if [ "$(id -u)" -ne 0 ]; then
+    echo "session-tool needs to be invoked as root" >&2
+    exit 1
+fi
+if [ -z "$(command -v busctl)" ]; then
+    echo "session-tool requires busctl" >&2
+    exit 1
+fi
+if [ $# -eq 0 ]; then
+	echo "usage: session-tool [-u USER] [--] <CMD>"
+	exit 1
+fi
+user=root
+while [ $# -gt 0 ]; do
+	case "$1" in
+		--)
+			shift
+			break
+			;;
+		-u)
+			if [ $# -eq 1 ]; then
+				echo "session-tool: option -u requires an argument" >&2
+				exit 1
+			fi
+			user="$2"
+			shift 2
+			;;
+		-*)
+			echo "session-tool: unsupported argument $1" >&2
+			exit 1
+			;;
+		*)
+			break
+			;;
+	esac
+done
+
+sed -i -e 's/mesg n .*true/tty -s \&\& mesg n/g' /root/.profile
+
+read -r uuid < /proc/sys/kernel/random/uuid
+unit_name="session-tool-$uuid.service"
+tmp_dir="/tmp/session-tool-$uuid"
+mkdir "$tmp_dir"
+mkfifo -m 0666 "$tmp_dir/dbus-monitor.pipe" "$tmp_dir/ready.pipe" "$tmp_dir/result.pipe" "$tmp_dir/stdin.pipe" "$tmp_dir/stdout.pipe" "$tmp_dir/stderr.pipe"
+trap 'rm -rf $tmp_dir' EXIT
+
+# Run dbus-monitor waiting for systemd JobRemoved signal. Process the output
+# with awk, forwarding the result of the service to a fifo.
+monitor_expr="type='signal', sender='org.freedesktop.systemd1', interface='org.freedesktop.systemd1.Manager', path='/org/freedesktop/systemd1', member='JobRemoved'"
+stdbuf -oL dbus-monitor --system --monitor "$monitor_expr" > "$tmp_dir/dbus-monitor.pipe" &
+dbus_monitor_pid=$!
+
+awk_expr="BEGIN {found=0; ready=0;} /member=NameAcquired/ { if (!ready) {ready=1; print ready > \"$tmp_dir/ready.pipe\"; close(\"$tmp_dir/ready.pipe\"); } } /   string \"$unit_name\"/ {found=1; print \"found service file\"; fflush(); next; }; /string \".*\"/ { if (found==1) {print \"found result\"; print \$2 > \"$tmp_dir/result.pipe\"; fflush(\"\"); close(\"$tmp_dir/result.pipe\"); exit; } }"
```

Comment:
could we use a heredoc for this or some other way that makes this snippet more readable? It's not super easy to follow being all in one line like this

Reply:
Yeah, a while after pushing I realized sh supports heretic as well.